apps/web/src/pages/Skills/components/RemoveDialog.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { Button, CardRepeater, Dialog } from "@gc-digital-talent/ui";
+import { commonMessages, formMessages } from "@gc-digital-talent/i18n";
+import { toast } from "@gc-digital-talent/toast";
+
+type RemoveDialogProps = {
+  onRemove: () => Promise<void>;
+  index: number;
+};
+
+const RemoveDialog = ({ onRemove, index }: RemoveDialogProps): JSX.Element => {
+  const intl = useIntl();
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+  const [isBusy, setIsBusy] = React.useState<boolean>(false);
+
+  const handleRemove = () => {
+    setIsBusy(true);
+    onRemove()
+      .then(() => {
+        toast.success(
+          intl.formatMessage({
+            defaultMessage: "Successfully updated skill",
+            id: "vMBiMV",
+            description: "Message displayed when a user updates a skill",
+          }),
+        );
+        setIsOpen(false);
+      })
+      .catch(() =>
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Error: updating skill failed",
+            id: "kfjmTt",
+            description:
+              "Message displayed to user after skill fails to be updated",
+          }),
+        ),
+      )
+      .finally(() => setIsBusy(false));
+  };
+
+  return (
+    <Dialog.Root open={isOpen} onOpenChange={(open) => setIsOpen(open)}>
+      <Dialog.Trigger>
+        <CardRepeater.Remove
+          aria-label={intl.formatMessage(formMessages.repeaterRemove, {
+            index,
+          })}
+        />
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>
+          {intl.formatMessage(commonMessages.remove)}
+        </Dialog.Header>
+        <Dialog.Body>
+          <p>
+            {intl.formatMessage({
+              defaultMessage: "You're about to remove this skill.",
+              id: "D0wyx7",
+              description: "First paragraph for remove skill dialog",
```

Comment:
Copy concern, should probably explain that it isn't deleting the skill from your profile but just from the showcase. I wasn't sure at first when reading the dialog and checked the mutation and library.

Reply:
Good idea!  I've update it.
[Change dialog string](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9592/commits/c75a8a1da868ded3936e8841993426cfcbee5cc3)